test/common/Serialize_unit_tests.cpp

```diff
@@ -0,0 +1,139 @@
+    EXPECT_EQ(nc::asset::GetBlobSize(expectedAsset), actualHeader.size);
+    EXPECT_STREQ("NONE", actualHeader.compressionAlgorithm);
+
+    EXPECT_EQ(expectedAsset.extents, actualAsset.extents);
+    EXPECT_EQ(expectedAsset.maxExtent, actualAsset.maxExtent);
+    ASSERT_EQ(expectedAsset.triangles.size(), actualAsset.triangles.size());
+
+    for(auto i = 0u; i < expectedAsset.triangles.size(); ++i)
+    {
+        const auto& expected = expectedAsset.triangles[i];
+        const auto& actual = actualAsset.triangles[i];
+        EXPECT_EQ(expected.a, actual.a);
+        EXPECT_EQ(expected.b, actual.b);
+        EXPECT_EQ(expected.c, actual.c);
+    }
+}
+
+TEST(SerializationTest, Mesh_roundTrip_succeeds)
+{
+    constexpr auto assetId = 1234ull;
+    const auto expectedAsset = nc::asset::Mesh{
+        .extents = nc::Vector3{-5.0f, 4.22f, 10.010101f},
+        .maxExtent = 10.010101f,
+        .vertices = std::vector<nc::asset::MeshVertex>{
+            nc::asset::MeshVertex{nc::Vector3::Splat(0.0f),
+                              nc::Vector3::Splat(1.0f),
+                              nc::Vector2::Splat(2.0f),
+                              nc::Vector3::Splat(3.0f),
+                              nc::Vector3::Splat(4.0f)},
+            nc::asset::MeshVertex{nc::Vector3::Splat(5.0f),
+                              nc::Vector3::Splat(6.0f),
+                              nc::Vector2::Splat(7.0f),
+                              nc::Vector3::Splat(9.0f),
+                              nc::Vector3::Splat(9.0f)},
+            nc::asset::MeshVertex{nc::Vector3::Splat(10.0f),
+                              nc::Vector3::Splat(11.0f),
+                              nc::Vector2::Splat(12.0f),
+                              nc::Vector3::Splat(13.0f),
+                              nc::Vector3::Splat(14.0f)}
+        },
+        .indices = std::vector<uint32_t>{
+            0, 1, 2,  1, 2, 0,  2, 0, 1
+        }
+    };
+
+    auto stream = std::stringstream{std::ios::in | std::ios::out | std::ios::binary};
+    nc::asset::Serialize(stream, expectedAsset, assetId);
+    const auto [actualHeader, actualAsset] = nc::asset::DeserializeMesh(stream);
+
+    EXPECT_STREQ("MESH", actualHeader.magicNumber);
+    EXPECT_EQ(assetId, actualHeader.assetId);
+    EXPECT_EQ(nc::asset::GetBlobSize(expectedAsset), actualHeader.size);
+    EXPECT_STREQ("NONE", actualHeader.compressionAlgorithm);
+
+    EXPECT_EQ(expectedAsset.extents, actualAsset.extents);
+    EXPECT_EQ(expectedAsset.maxExtent, actualAsset.maxExtent);
+    ASSERT_EQ(expectedAsset.vertices.size(), actualAsset.vertices.size());
+    ASSERT_EQ(expectedAsset.indices.size(), actualAsset.indices.size());
+
+    for(auto i = 0u; i < expectedAsset.vertices.size(); ++i)
+    {
+        const auto& e = expectedAsset.vertices[i];
+        const auto& a = actualAsset.vertices[i];
+        EXPECT_EQ(e, a);
+    }
+
+    // EXPECT_TRUE(std::equal(expectedAsset.vertices.cbegin(),
```

Comment:
Does this commented out code need to hang around?

Reply:
Nope!